eos_data_distribution/ndn/file.py

```diff
@@ -274,6 +264,44 @@ def _on_complete(self):
         os.unlink(self._sgt_filename)
         super(FileConsumer, self)._on_complete()
 
+    def _create_files(self, filename):
+        self._filename = filename
+        mkdir_p(os.path.dirname(filename))
+        self._part_filename = '%s.part' % (filename, )
+        self._part_fd = os.open(
+            self._part_filename, os.O_CREAT | os.O_WRONLY | os.O_NONBLOCK, 0o600)
+        self._sgt_filename = '%s.sgt' % (filename, )
+        self._sgt_fd = os.open(
+            self._sgt_filename, os.O_CREAT | os.O_RDWR, 0o600)
```

Comment:
Technically, if one of the I/O operations which is not the first one raises an exception, the directory will be left in an unknown state (some of the files could exist; others won’t). It would be good to clean that up so the whole function is effectively atomic wrt I/O. This could be done as a follow-up commit, as the code here is simply moved from elsewhere.